### PR TITLE
[webhook] add Vault namespaces support

### PR DESCRIPTION
**Pull request by anoncam**
fixes: https://github.com/banzaicloud/bank-vaults/issues/1347

### CODE REVIEW

**Review by bonifaido — APPROVED** [review]
LGMT! Thank you very much!

Could you please fix the unit test as well?

**Review by pbalogh-sa — APPROVED** [review]
Thanks! LGTM

**Comment by anoncam** [comment]
> LGMT! Thank you very much!
> 
> Could you please fix the unit test as well?

Yes I can, can you point me to the items I need to fix lol?  I didn't write any unit tests which is likely the problem haha.

**Comment by bonifaido** [comment]
If you check the CI, you can find the related errors:

https://github.com/banzaicloud/bank-vaults/pull/1361/checks?check_run_id=2893575173

**Comment by anoncam** [comment]
Got it. I'm a bit stupid apologies for not having this ready to go on first pass.

**Comment by anoncam** [comment]
I think this is ready now ... lol I'm clearly not a golang dev -- but ran the test locally and it was `ok`

**Comment by anoncam** [comment]
@bonifaido can you review so this can be merged?

File: pkg/webhook/config.go [hunk 1]
```diff
@@ -397,5 +404,6 @@ func SetConfigDefaults() {
 	viper.SetDefault("VAULT_ENV_MEMORY_REQUEST", "")
 	viper.SetDefault("VAULT_ENV_CPU_LIMIT", "")
 	viper.SetDefault("VAULT_ENV_MEMORY_LIMIT", "")
+	viper.SetDefault("VAULT_NAMESPACE", "root")
```
Comment (bonifaido) [inline pkg/webhook/config.go:407]: Please make this default to `""`.

Comment (anoncam) [reply pkg/webhook/config.go:407]: copy will do

Comment (anoncam) [reply pkg/webhook/config.go]: `root` is the default namespace for any vault

Comment (anoncam) [reply pkg/webhook/config.go]: but it is fixed

Comment (anoncam) [reply pkg/webhook/config.go]: @bonifaido what am I missing?  Acceptance failed this time

Comment (anoncam) [reply pkg/webhook/config.go]: I'm starting to feel like a total moron

**Review by bonifaido — APPROVED** [review]
Seems to be working now! :)

File: pkg/webhook/config.go (hunk 1 above)
Comment (bonifaido) [reply pkg/webhook/config.go]: You don't have to, it was just a flaky test! Everything is working now, thank you for getting this through the repo!

**Pull request merged**